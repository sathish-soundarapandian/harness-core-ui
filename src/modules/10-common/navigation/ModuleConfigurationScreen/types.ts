/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { Entry } from 'contentful'
import type { IconName } from '@harness/icons'
import type { ModuleName } from 'framework/types/ModuleName'

export interface LottieContent {
  data: Record<any, any>
}

export interface CenterAlignedImageDesc {
  primaryText?: string
  secondoryText?: string
  imageUrl: string
}

export interface ModuleInfo {
  label: string
  icon: IconName
}

export interface ContentfulModuleResponse {
  identifier: ModuleName
  label: string
  data: Entry<CenterAlignedImageDesc | LottieContent>[]
}
