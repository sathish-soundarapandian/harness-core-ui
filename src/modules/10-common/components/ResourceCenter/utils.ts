/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import moment from 'moment'
import type { UserInfo } from 'services/cd-ng'

export const ON_PREM_RELEASE_NODE_LINK = 'https://ngdocs.harness.io/article/556wy85kbo-harness-on-prem-release-notes'
export const SAAS_RELEASE_NODE_LINK = 'https://ngdocs.harness.io/article/7zkchy5lhj-harness-saa-s-release-notes-2022'

export const HARNESS_SEARCH_LINK = 'https://harness.io/search/'
export const HARNESS_UNIVERISITY_LINK = 'https://university.harness.io/page/job-role-certifications'
export const HARNESS_COMMUNITY_LINK = 'https://community.harness.io/'
export const HARNESS_COMMUNITY_SLACK_LINK =
  'https://join.slack.com/t/harnesscommunity/shared_invite/zt-y4hdqh7p-RVuEQyIl5Hcx4Ck8VCvzBw'
export const HARNESS_DOCS_LINK = 'https://docs.harness.io/'
export const HARNESS_API_DOCS_LINK = 'https://apidocs.harness.io/'
export const HARNESS_TUTORIALS = 'https://developer.harness.io/tutorials/get-started'
export const SITE_STATUS_LINK = 'https://status.harness.io/'

export const timestamp = moment.now()
export const HARNESS_SUPPORT_LINK =
  '/sso.html?action=login&brand_id=114095000394&locale_id=1&return_to=https%3A%2F%2Fsupport.harness.io%2Fhc%2Fen-us%2Frequests&src=zendesk&timestamp=' +
  timestamp

export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`

export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(HARNESS_SUPPORT_LINK)
}

export function getReleaseNodeLink(): string {
  switch (window.deploymentType) {
    case 'COMMUNITY':
    case 'ON_PREM': {
      return ON_PREM_RELEASE_NODE_LINK
    }
    default:
      return SAAS_RELEASE_NODE_LINK
  }
}
export const openFileATicket = (
  e: React.MouseEvent<Element, MouseEvent>,
  currentUserInfo: UserInfo,
  setShow?: React.Dispatch<React.SetStateAction<boolean>>
): void => {
  e.stopPropagation()
  e.preventDefault()
  window.Saber.do?.('set_options', {
    feedback_values: {
      Email: currentUserInfo.email // set default value for email field
    }
  })
  setShow?.(false)
  window.Saber.do?.('open')
}
export const openWhatsNew = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(WHATS_NEW_LINK)
}
export const openEarlyAccess = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(EARLY_ACCESS_LINK)
}
