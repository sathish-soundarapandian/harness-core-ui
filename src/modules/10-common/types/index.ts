import type { QlceViewTimeGroupType, QlceViewFilterInput, QlceViewFieldInputInput } from 'services/ce/services'

export interface TimeRangeFilterType {
  to: string
  from: string
}

export enum TimeRangeType {
  'LAST_7' = 'LAST_7',
  'LAST_30' = 'LAST_30'
}

export enum NodepoolTimeRangeType {
  'LAST_DAY' = 'LAST_DAY',
  'LAST_7' = 'LAST_7',
  'LAST_30' = 'LAST_30'
}

export type setFiltersFn = (newFilters: QlceViewFilterInput[]) => void
export type setAggregationFn = (newAgg: QlceViewTimeGroupType) => void
export type setGroupByFn = (groupBy: QlceViewFieldInputInput) => void
export type setTimeRangeFn = (timeRange: TimeRangeFilterType) => void
