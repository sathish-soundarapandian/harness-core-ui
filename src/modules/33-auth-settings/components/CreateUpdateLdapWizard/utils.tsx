/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Text, Icon, Layout } from '@harness/uicore'
import { Color, FontVariation } from '@harness/design-system'
import React, { ReactElement } from 'react'
import { useStrings } from 'framework/strings'
import type { LdapConnectionSettings, ResponseMessage } from 'services/cd-ng'
import { ErrorHandler } from '@common/components/ErrorHandler/ErrorHandler'

export interface RawLdapConnectionSettings {
  accountId?: string
  bindDN?: string
  bindPassword?: string
  bindSecret?: string[]
  connectTimeout?: number
  host: string
  maxReferralHops?: number
  passwordType?: string
  port?: number
  referralsEnabled?: boolean
  responseTimeout?: number
  sslEnabled?: boolean
  useRecursiveGroupMembershipSearch?: boolean
}

export const updateLDAPConnectionSettingsFormData = (
  values: RawLdapConnectionSettings | undefined,
  accountId: string,
  maxReferralHops: number | undefined
): LdapConnectionSettings | void => {
  if (!values) {
    return
  }

  const updatedValues = { ...values }

  if (values['referralsEnabled']) {
    if (values['maxReferralHops']) {
      updatedValues['maxReferralHops'] = Number(values['maxReferralHops'])
    } else {
      updatedValues['maxReferralHops'] = maxReferralHops
    }
  } else {
    delete updatedValues['maxReferralHops']
  }

  if (!values['sslEnabled']) {
    updatedValues['sslEnabled'] = false
  }

  delete updatedValues['bindSecret']

  return {
    ...updatedValues,
    passwordType: 'INLINE',
    port: Number(values['port']),
    connectTimeout: Number(values['connectTimeout']),
    responseTimeout: Number(values['responseTimeout']),
    accountId: accountId
  }
}

export const QueryTestSuccessMsg: React.FC<{ message: string }> = ({ message }) => {
  return (
    <>
      <Icon name="success-tick" size={16} margin={{ right: 'xsmall' }} />
      <Text font={{ variation: FontVariation.FORM_MESSAGE_SUCCESS }} color={Color.GREEN_800}>
        {message}
      </Text>
    </>
  )
}

export const QueryTestFailMsgs: React.FC<{ errorMessages: ResponseMessage[] }> = ({ errorMessages }) => {
  return <ErrorHandler responseMessages={errorMessages} />
}

export const QueryStepTitle: React.FC<{ stepTitle?: string }> = ({ stepTitle }) => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal margin={{ bottom: 'medium' }} style={{ alignItems: 'center' }}>
      <Text font={{ variation: FontVariation.H4 }} margin={{ right: 'small' }}>
        {stepTitle}
      </Text>
      <Text font={{ weight: 'light' }}>{getString('titleOptional')}</Text>
    </Layout.Horizontal>
  )
}

export const QueryFormTitle: React.FC<{ title: string }> = ({ title }) => {
  return (
    <>
      <Text font={{ variation: FontVariation.H5 }}>
        <Icon name="chevron-down" color={Color.PRIMARY_6} margin={{ right: 'small' }} />
        {title}
      </Text>
    </>
  )
}

export const getErrorMessageFromException = (e: any, fallbackMessage?: string): ResponseMessage[] => {
  const hasResponseMessages = e.data?.responseMessages && e.data?.responseMessages.length > 0
  return hasResponseMessages
    ? e.data?.responseMessages
    : [{ level: 'ERROR', message: e.data?.message || e.message || fallbackMessage }]
}

export const DEFAULT_LDAP_SYNC_CRON_EXPRESSION = '0 0/15 * 1/1 * ? *' // Every 15 minutes

export const getDateIterationsList = (dates: number[]): ReactElement => {
  const datesHTML = dates.map(date => {
    const dateObject = new Date(date)
    return <li key={date}>{dateObject.toString()}</li>
  })

  return <ul>{datesHTML}</ul>
}
