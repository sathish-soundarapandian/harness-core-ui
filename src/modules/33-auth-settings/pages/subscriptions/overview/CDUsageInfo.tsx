/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout, PageError } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { useGetUsageAndLimit } from '@common/hooks/useGetUsageAndLimit'
import { ContainerSpinner } from '@common/components/ContainerSpinner/ContainerSpinner'
import { ModuleName } from 'framework/types/ModuleName'
import type { ModuleLicenseDTO, CDModuleLicenseDTO } from 'services/cd-ng'
import UsageInfoCard, { ErrorContainer } from './UsageInfoCard'

interface SubscriptionUsageProps {
  module: ModuleName
  licenseData: ModuleLicenseDTO
}
const ActiveInstanceCard: React.FC<{ subscribedIns: number; activeIns: number; displayName?: string }> = ({
  subscribedIns,
  activeIns
}) => {
  const { getString } = useStrings()

  const leftHeader = getString('common.subscriptions.usage.srvcInst')
  const tooltip = getString('common.subscriptions.usage.cdSITooltip')
  const rightHeader = getString('common.subscriptions.usage.last30days')
  const hasBar = true
  const leftFooter = getString('common.subscribed')
  const rightFooter = getString('common.usage')
  const props = {
    subscribed: subscribedIns,
    usage: activeIns,
    leftHeader,
    tooltip,
    rightHeader,
    hasBar,
    leftFooter,
    rightFooter
  }
  return <UsageInfoCard {...props} />
}

const ActiveServiceCard: React.FC<{ subscribedIns: number; activeIns: number; displayName?: string }> = ({
  subscribedIns,
  activeIns
}) => {
  const { getString } = useStrings()

  const leftHeader = getString('common.subscriptions.usage.serviceLicenses')
  const tooltip = getString('common.subscriptions.usage.cdServiceTooltip')
  const rightHeader = getString('common.subscriptions.usage.last30days')
  const hasBar = true
  const leftFooter = getString('common.subscribed')
  const rightFooter = getString('common.usage')
  const props = {
    subscribed: subscribedIns,
    usage: activeIns,
    leftHeader,
    tooltip,
    rightHeader,
    hasBar,
    leftFooter,
    rightFooter
  }
  return <UsageInfoCard {...props} />
}

const CDUsageInfo: React.FC<SubscriptionUsageProps> = props => {
  const { limitData, usageData } = useGetUsageAndLimit(ModuleName.CD)
  const isLoading = limitData.loadingLimit || usageData.loadingUsage
  const { licenseData } = props

  const licenseDataInfo = licenseData as CDModuleLicenseDTO
  if (isLoading) {
    return <ContainerSpinner />
  }

  const { usageErrorMsg, refetchUsage, usage } = usageData
  const { limitErrorMsg, refetchLimit } = limitData

  if (usageErrorMsg) {
    return (
      <ErrorContainer>
        <PageError message={usageErrorMsg} onClick={() => refetchUsage?.()} />
      </ErrorContainer>
    )
  }

  if (limitErrorMsg) {
    return (
      <ErrorContainer>
        <PageError message={limitErrorMsg} onClick={() => refetchLimit?.()} />
      </ErrorContainer>
    )
  }
  if (licenseDataInfo.cdLicenseType === 'SERVICE_INSTANCES') {
    return (
      <Layout.Horizontal spacing="large">
        <ActiveInstanceCard
          subscribedIns={licenseDataInfo.workloads || 0}
          activeIns={usage?.cd?.activeServiceInstances?.count || 0}
        />
      </Layout.Horizontal>
    )
  } else {
    return (
      <Layout.Horizontal spacing="large">
        <ActiveServiceCard
          subscribedIns={licenseDataInfo.workloads || 0}
          activeIns={usage?.cd?.serviceLicenses?.count || 0}
        />
      </Layout.Horizontal>
    )
  }
}

export default CDUsageInfo
