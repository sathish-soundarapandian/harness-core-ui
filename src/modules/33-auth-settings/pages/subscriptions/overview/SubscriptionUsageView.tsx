/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout } from '@harness/uicore'
import { ModuleName } from 'framework/types/ModuleName'
import type { ModuleLicenseDTO } from 'services/cd-ng'
import CDUsageTable from './CDUsageTable'
import CIUsageTable from './CIUsageTable'

interface SubscriptionUsageViewProps {
  module: ModuleName
  licenseData?: ModuleLicenseDTO
}

const getModuleUsagesTable = (props: SubscriptionUsageViewProps): React.ReactElement | undefined => {
  switch (props.module) {
    case ModuleName.CI:
      return <CIUsageTable {...props} />
    case ModuleName.CD:
      return <CDUsageTable {...props} />
    default:
      return undefined
  }
}

const SubscriptionUsageView: React.FC<SubscriptionUsageViewProps> = props => {
  const usageModuleTable = getModuleUsagesTable(props)
  return usageModuleTable ? (
    <Layout.Vertical spacing="xxlarge" flex={{ alignItems: 'baseline', justifyContent: 'space-between' }}>
      {usageModuleTable}
    </Layout.Vertical>
  ) : (
    <></>
  )
}

export default SubscriptionUsageView
