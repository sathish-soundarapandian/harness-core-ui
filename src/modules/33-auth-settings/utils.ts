/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import { get } from 'lodash-es'
import { CurrencyType } from '@common/constants/SubscriptionTypes'

export function getAmountInCurrency(currency: CurrencyType, amount: number): string {
  switch (currency) {
    case CurrencyType.USD: {
      return amount.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
      })
    }
  }

  return ''
}

// dividing the actual amount by 100 as amount numbers are in cents
export function getDollarAmount(amount?: number, isYearly?: boolean): number {
  if (isYearly) {
    return (amount || 0) / 1200
  }
  return (amount || 0) / 100
}

export const getErrorMessage = (error: any): string =>
  get(error, 'data.error', get(error, 'data.message', error?.message))
