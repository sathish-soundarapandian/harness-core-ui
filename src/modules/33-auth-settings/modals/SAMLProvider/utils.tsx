/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { AuthenticationMechanisms } from '@rbac/utils/utils'
import type { UseStringsReturn } from 'framework/strings'
export interface FormValues {
  displayName: string
  friendlySamlName?: string
  authorizationEnabled: boolean
  groupMembershipAttr: string
  entityIdEnabled: boolean
  entityIdentifier: string
  logoutUrl?: string
  clientSecret?: string
  clientId?: string
  samlProviderType?: Providers
  enableClientIdAndSecret: boolean
  jitEnabled?: boolean
  jitValidationKey?: string
  jitValidationValue?: string
}

export enum Providers {
  AZURE = 'AZURE',
  OKTA = 'OKTA',
  ONE_LOGIN = 'ONELOGIN',
  OTHER = 'OTHER'
}

export interface SAMLProviderType {
  value: Providers
  label: string
  icon: IconName
}

export const createFormData = (data: FormValues): FormData => {
  const formData = new FormData()
  formData.set('displayName', data.displayName)
  formData.set('authorizationEnabled', JSON.stringify(data.authorizationEnabled))
  formData.set('groupMembershipAttr', data.groupMembershipAttr)
  formData.set('ssoSetupType', AuthenticationMechanisms.SAML)

  if (data.friendlySamlName) {
    formData.set('friendlySamlName', data.friendlySamlName)
  }

  if (data.logoutUrl) {
    formData.set('logoutUrl', data.logoutUrl)
  }

  if (data.samlProviderType) {
    formData.set('samlProviderType', data.samlProviderType)
  }

  if (data.samlProviderType === Providers.AZURE) {
    if (data.authorizationEnabled && data.clientId && data.enableClientIdAndSecret) {
      formData.set('clientId', data.clientId)
    }
    if (data.authorizationEnabled && data.clientSecret && data.enableClientIdAndSecret) {
      formData.set('clientSecret', data.clientSecret)
    }
  }
  if (data.entityIdEnabled && data.entityIdentifier) {
    formData.set('entityIdentifier', data.entityIdentifier)
  }

  if (data.jitEnabled) {
    formData.set('jitEnabled', data.jitEnabled.toString())
    // jit validation key & value are optional
    if (data.jitValidationKey && data.jitValidationValue) {
      formData.set('jitValidationKey', data.jitValidationKey)
      formData.set('jitValidationValue', data.jitValidationValue)
    }
  }

  const file = (data as any)?.files?.[0]
  file && formData.set('file', file)

  return formData
}

export const getSelectedSAMLProvider = (
  selected: SAMLProviderType | undefined,
  getString: UseStringsReturn['getString']
): string => {
  if (selected && selected?.value !== Providers.OTHER) {
    return selected.label
  }
  return getString('authSettings.SAMLProvider')
}
