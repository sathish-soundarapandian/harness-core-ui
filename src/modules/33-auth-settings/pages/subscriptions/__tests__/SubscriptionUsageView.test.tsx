/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { ModuleName } from 'framework/types/ModuleName'
import SubscriptionUsageView from '../overview/SubscriptionUsageView'
import { licenseData } from './CDUsageInfo.test'

const featureFlags = {
  CVNG_ENABLED: true,
  CING_ENABLED: true,
  CENG_ENABLED: true,
  CFNG_ENABLED: true,
  CHAOS_ENABLED: true
}

describe('Subscriptions graph view undefined module flow  ', () => {
  test('it renders the subscriptions page with no data service table', async () => {
    const { container } = render(
      <TestWrapper
        defaultAppStoreValues={{ featureFlags }}
        defaultLicenseStoreValues={{
          licenseInformation: {
            CF: { edition: 'FREE', status: 'ACTIVE' }
          }
        }}
      >
        <SubscriptionUsageView licenseData={licenseData} module={ModuleName.CF} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
