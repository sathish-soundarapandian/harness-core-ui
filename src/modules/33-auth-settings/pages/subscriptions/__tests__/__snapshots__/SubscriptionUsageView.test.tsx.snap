// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CDUsageInfo CDUsageInfo 1`] = `
<div>
  <div
    class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
  >
    <div
      class="bp3-card bp3-elevation-0 Card--card innerCard"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline StyledProps--flex-justifyContent-space-between"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--color StyledProps--color-grey700"
            >
              common.subscriptions.usage.serviceLicenses
            </p>
            <span
              class="bp3-popover-wrapper"
            >
              <span
                class="bp3-popover-target"
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="info"
                  tabindex="0"
                >
                  <svg
                    height="15"
                    viewBox="0 0 12 12"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96zm1.846-7.352v1.03h-1.01v-1.03z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--color StyledProps--color-grey500"
          >
            common.subscriptions.usage.last30days
          </p>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-large StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
        >
          0
        </p>
        <div
          class="StyledProps--font StyledProps--main bar StyledProps--background-grey100 StyledProps--border StyledProps--margin-bottom-small"
          style="width: 100%; height: 10px;"
        >
          <div
            class="StyledProps--font StyledProps--main StyledProps--background-primary6 StyledProps--border"
            style="width: 0%; height: 100%;"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
          >
            100
             
            common.subscribed
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--color"
          >
            0
            % 
            common.usage
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Subscriptions graph view undefined module flow   it renders the subscriptions page with no data service table 1`] = `<div />`;
